### PR TITLE
handle Katello-style errors in fail_from_exception

### PR DESCRIPTION
**Pull request by evgeni**
Fixes: #546

### CODE REVIEW

File: plugins/module_utils/foreman_helper.py [hunk 1]
```diff
@@ -525,7 +525,11 @@ def fail_from_exception(self, exc, msg):
         fail = {'msg': msg}
         if isinstance(exc, requests.exceptions.HTTPError):
             try:
-                fail['error'] = exc.response.json()['error']
+                response = exc.response.json()
```
Comment (evgeni) [inline plugins/module_utils/foreman_helper.py]: this still can raise an exception and thus fuck up the traceback, improvements welcome

Comment (mdellweg) [reply plugins/module_utils/foreman_helper.py]: You mean when the response is not json deserializable?

Comment (evgeni) [reply plugins/module_utils/foreman_helper.py]: Yes, response.json will throw a valueerror

Comment (mdellweg) [reply plugins/module_utils/foreman_helper.py]: For the occasional 503?
But there is a `try` to catch that. I think we are fine.

Comment (evgeni) [reply plugins/module_utils/foreman_helper.py]: Oh, i wasn't verbose enough. Sorry.

Fail json will output a traceback if there is one (without us doing anything). Now if a ValueError happens here, the traceback will be for that and not from the exception we're handling.

**Comment by evgeni** [comment]
merging as this is still better then before.

**Pull request merged**